plugins/woocommerce/includes/wc-deprecated-functions.php

```diff
@@ -137,6 +137,27 @@ function wc_deprecated_argument( $argument, $version, $message = null ) {
 	}
 }
 
+/**
+ * Call with script handle to warn about deprecated scripts. If WP_DEBUG 
+ * is enabled an error will be thrown.
+ * 
+ * @since 7.5.0
+ * @param string $handle
+ * @param string $version
+ * @param string $replacement
+ * @param string $message
+ */
+function wc_deprecated_script( $handle, $version, $replacement = null, $message = '' ) {
+	$error_message = "The $handle script is deprecated since version $version. $message";
+	$error_message .= $replacement ? " Use $replacement instead." : '';
+	
+	if ( wp_doing_ajax() || WC()->is_rest_api_request() ) {
+		error_log( $error_message );
+	} else if ( WP_DEBUG ) {
```

Comment:
we need to use a `defined` check here, in PHP < 7.4, this will be assumed to be `WP_DEBUG` string which will be evaluated to true

Reply:
TIL thanks I'll fix and add unit test!